client/src/components/InviteIn.js

```diff
@@ -1,8 +1,98 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import { Typography } from '@material-ui/core';
+import CheckIcon from '@material-ui/icons/Check';
+import ClearIcon from '@material-ui/icons/Clear';
+import { makeStyles } from '@material-ui/core/styles';
 
-const InviteIn = (props) => {
-  return <div>InviteIn</div>;
+const useStyles = makeStyles((theme) => ({
+  inviteItem: {
+    height: 100,
+    width: '85%',
+    display: 'flex',
+    justifyContent: 'space-between',
+    alignItems: 'center',
+    borderBottom: 'solid 1px #ddd',
+    '& div': {
+      display: 'flex',
+      alignItems: 'center',
+    },
+  },
+  sideBarImg: {
+    borderRadius: '100%',
+    overflow: 'hidden',
+    width: '40px',
+    marginRight: theme.spacing(1),
+  },
+  button: {
+    background: 'none',
+    outline: 'none',
+    border: 'none',
+    cursor: 'pointer',
+  },
+  icon: {
+    fontSize: 30,
+  },
+  check: {
+    color: theme.palette.primary.blue,
+    '&:hover': {
+      color: theme.palette.secondary.green,
+    },
+  },
+  cross: {
+    '&:hover': {
+      color: 'red',
+    },
+  },
+}));
+
+const InviteIn = ({ invite, invitesIn, setInvitesIn }) => {
+  const { avatar, name } = invite;
+  const classes = useStyles();
+
+  const handleApprove = () => {
+    //remove invite from invitesIn
+    //change status to approved
+    //add to contacts
+
+    invite.status = 'approved';
+    let invitesInCopy = [...invitesIn];
+
+    let filteredInvites = invitesInCopy.filter((request) => {
+      return request.email !== invite.email;
+    });
+    console.log('invite:', invite);
+    setInvitesIn(filteredInvites);
```

Comment:
So best practice in React is to never modify from the state of a parent component from a child component. So rather than calling setInvitesIn here, we would want a function in Invite.js that handles everything this function does at that higher level. Also, you can then pass the index of this invite so that you don't have to filter through the invites to find by email

Reply:
got it.  Do you mean use indexOf and splice the target out of the array?

Reply:
What I mean is that when we loop through invites to create each InviteIn component, we can pass it its index as a prop. Then we can just pass that prop as a parameter for this function, and splice it out with that, no need to loop through all of the invites at all

Reply:
function in Invites component

```
  const handleApproveRequest = (invite) => {
    //change status of invite and save it
    invite.status = 'approved';
    invites.splice(invite.index, 1);
  }
```;

Handling the mapping and index like you suggested:

```
invites.pendingInvitesIn.map((invite, index) => {
                invite.index = index;
                return (
                  <InviteIn
                    key={invite.referrer}
                    invite={invite}
                    invitesIn={invites.pendingInvitesIn}
                    handleApproveRequest={handleApproveRequest}
                    handleRejectRequest={handleRejectRequest}
                  />
                );
              }
```